README.md

```diff
@@ -25,14 +25,14 @@ The symbolic automaton generalizes the classic automaton's transition to a predi
 ```
 
-This has implications that we could explore the automaton without requiring to know the full alphabet and in fact the alphabet can now be infinite, which sacrificing any decidability properties. Unfortunately this does push some complexity onto the transitions, which can grow into `if` expressions with exponential complexity.
+This has the implication that we could explore the automaton without knowing the full alphabet and in fact the alphabet can now be infinite, without sacrificing any decidability properties. Unfortunately this does push some complexity onto the transitions, which can grow into `if` expressions with exponential complexity.
 
-- A basic introduction to symbolic automaton can be found in: [Symbolic Boolean Derivatives - Caleb Stanford, Margus Veanes and Nikolaj Bjorner](https://www.cis.upenn.edu/~castan/doc/2021/PLDI21.pdf) they also show how derivatives can be used to add better regex support to z3, which is very interesting, but adds complexities that aren't necessarily applicable to us.
-- A larger overview of how symbolic automaton fit into the larger computer science field including tree transducers and more, see ["The Power of Symbolic Automata and Transducers" Loris D’Antoni | CAV 2017](https://www.youtube.com/watch?v=ca9IF-7nSOA)
+- A basic introduction to symbolic automaton can be found in [Symbolic Boolean Derivatives - Caleb Stanford, Margus Veanes and Nikolaj Bjorner](https://www.cis.upenn.edu/~castan/doc/2021/PLDI21.pdf). They also show how derivatives can be used to add better regex support to z3, which is very interesting, but adds complexities that aren't necessarily applicable to us.
+- For a larger overview of how symbolic automata fit into the larger computer science field including tree transducers and more, see ["The Power of Symbolic Automata and Transducers" Loris D’Antoni | CAV 2017](https://www.youtube.com/watch?v=ca9IF-7nSOA)
 
-In the final implementation of the algorithm, these predicates can be a set of nested function, for example: `and(ge($int, 10), lt($int, 20))`, which we can also explore, where we could prove that evaluation during compilation of the constant parts are equivalent to evaluation at runtime.
+In the final implementation of the algorithm, these predicates can be a set of nested functions, for example: `and(ge($int, 10), lt($int, 20))`. This is also something we can explore, and we could prove that evaluation during compilation of the constant parts is equivalent to evaluation at runtime.
 
-The reason to explore symbolic automaton first, is because symbolic derivatives can technically use trees as its input alphabet and so provide a quick win, but also we will also heavily use the `if` expressions in our final VPA implementation.
+The reason to explore symbolic automaton first, is because symbolic automata can technically use trees as their input alphabet and so provide a quick win, but also we will also heavily use the `if` expressions in our final VPA implementation.
```

Comment:
Is the `derivatives` -> `automata` change correct?

Reply:
their is good